instrumentation/resources/library/src/main/java/io/opentelemetry/instrumentation/resources/ContainerResource.java

```diff
@@ -77,19 +77,83 @@ public static Resource get() {
   // Exists for testing
   static class Filesystem {
 
+    private static final Logger logger = Logger.getLogger(Filesystem.class.getName());
+
     boolean isReadable(Path path) {
       return Files.isReadable(path);
     }
 
     @MustBeClosed
     Stream<String> lines(Path path) throws IOException {
-      return Files.lines(path);
+
+      String osName = getOsName();
+      if (!(osName.equalsIgnoreCase("z/OS") || osName.equalsIgnoreCase("OS/390"))) {
```

Comment:
could switch the if and else parts to get rid of the negation

Reply:
I put it this way around because making the default case the base one feels right, but I sense you have different tastes and its your codebase so I'll change it. (I'll test the updated version in our ZOS environment before pushing of course, which takes a bit of time)